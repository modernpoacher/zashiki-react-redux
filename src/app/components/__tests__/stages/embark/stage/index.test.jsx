import React from 'react'
import renderer from 'react-test-renderer'

import Stage from '#app/components/stages/embark/stage'

const MOCK_DEFINITION = {
  meta: {
    uri: '#/'
  },
  elements: {
    field: {
      id: '#/'
    }
  }
}

const MOCK_RESPONSE = {
  '#/': 'MOCK VALUE'
}

describe('#app/components/stages/embark/stage', () => {
  describe('Always', () => {
    it('renders', () => {
      const component = (
        <Stage
          definition={MOCK_DEFINITION}
          response={MOCK_RESPONSE}
          errors={[]}
          onChange={jest.fn()}
          onSubmit={jest.fn()}
        />
      )

      expect(renderer.create(component).toJSON())
        .toMatchSnapshot()
    })
  })
})
