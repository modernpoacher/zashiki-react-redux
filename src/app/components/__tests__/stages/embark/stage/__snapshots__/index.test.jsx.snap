// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#zashiki-react-redux/app/components/stages/embark/stage Always renders 1`] = `
<div
  className="embark"
>
  <form
    action="/embark-stage"
    method="POST"
    onSubmit={[Function]}
  >
    <div
      className="cog text"
    >
      <input
        className="input text"
        disabled={false}
        id="#/"
        name="#/"
        onChange={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
    <button
      type="submit"
    >
      Continue
    </button>
  </form>
</div>
`;
